base/compiler/tfuncs.jl

```diff
@@ -1230,15 +1230,16 @@ function _fieldtype_tfunc(@nospecialize(s), exact::Bool, @nospecialize(name))
         return Const(ft)
     end
 
-    exactft = exact || (!has_free_typevars(ft) && u.name !== Tuple.name)
+    exactft = !has_free_typevars(ft)
     ft = rewrap_unionall(ft, s)
-    if exactft
+    if exact && s_uniquerep
+        return Const(ft)
+    elseif exact || (exactft && u.name !== Tuple.name)
         if hasuniquerep(ft)
```

Comment:
Can we instead of any of the the other changes just improve this line to:
```suggestion
        if exact || hasuniquerep(ft)
```

Reply:
No, `exact` is the assertion that the type is `==` at runtime, but we need `===`.